jerry-main/cli.h

```diff
@@ -0,0 +1,151 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CLI_H
+#define CLI_H
+
+#include <string.h>
+
+/*
+ * Types for CLI
+ */
+
+/**
+ * Command line option definition
+ */
+typedef struct
+{
+  int id; /**< unique ID of the option (CLI_OPT_END, CLI_OPT_POSITIONAL, or anything >= 0) */
+  const char *opt; /**< short option variant (in the form of "-x") */
+  const char *longopt; /**< long option variant (in the form of "--xxx") */
+  int argc; /**< number of arguments of the option */
+  const char *meta; /**< name(s) of the argument(s) of the option, for display only */
+  int quant; /**< quantifier of the option (CLI_QUANT_{Q,A,P,1} for ?,*,+,1), for display only */
+  const char *help; /**< descriptive help message of the option */
+} cli_opt_t;
+
+/**
+ * Quantifiers of command line options
+ */
+typedef enum
+{
+  CLI_QUANT_Q, /**< ? (Question mark: optional, zero or one) */
+  CLI_QUANT_A, /**< * (Asterisk, star: zero or one or more) */
+  CLI_QUANT_P, /**< + (Plus sign: one or more) */
+  CLI_QUANT_1  /**< 1 (One: one) */
```

Comment:
That's not the same, that's what `.argc` is for. `.quant` describes whether and option (together with its argument(s)) is optional or not (or how many times it should appear on the command line). E.g., `--exec-snapshot snapfile1 --exec-snapshot snapfile2 --exec-snapshot snapfile3` is a valid use, that's why we have `[--exec-snapshot FILE]...` in the usage. For the snapshot merging patch, another option is to be added, used as `--exec-snapshot-func snapfile1 index1` IIRC, which means that it has two arguments, but the option together with its two arguments may also appear arbitrarily many times on the command line (`.argc = 2, .meta = "FILE NUM", .quant = CLI_QUANT_A`), displayed in usage as `[--exec-snapshot-func FILE NUM]...`.